app/main.py

```diff
@@ -1,8 +1,29 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        if self not in Person.people:
+            Person.people[self.name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    person_list = []
+    for person in people:
+        person_list.append(Person(name=person["name"], age=person["age"]))
+
+    for person in people:
+        current_person = Person.people[person["name"]]
+
+        if "wife" in person and person["wife"]:
+            wife_instance = Person.people.get(person["wife"])
+            if wife_instance:
+                current_person.wife = wife_instance
```

Comment:
You could avoid using nested if condition (because you've created all instances in the previous loop) and write if condition more efficiently, e.g.
```suggestion
        if wife_name := person.get("wife"):
            current_person.wife = Person.people[wife_name]
```

Reply:
Rewrite if condition for husband as well